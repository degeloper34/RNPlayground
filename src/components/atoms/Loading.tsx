import {ActivityIndicator, View, StyleSheet} from "react-native";

export function Loading({
  isLoadingActive,
}: {
  isLoadingActive: boolean | undefined;
}) {
  if (isLoadingActive) {
    return (
      <View style={styles.indicatorView}>
        <ActivityIndicator size="large" color="#F1D00A" />
      </View>
    );
  } else {
    return null;
  }
}

const styles = StyleSheet.create({
  indicatorView: {
    position: "absolute",
    justifyContent: "center",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    backgroundColor: "rgba(0,0,0,0.5)",
  },
});
