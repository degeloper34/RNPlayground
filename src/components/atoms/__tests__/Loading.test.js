import renderer from "react-test-renderer";
import React from "react";
import {Loading} from "../Loading";

describe("Loading", () => {
  it("renders correctly with loading enable", () => {
    const component = renderer.create(<Loading isLoadingActive />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with loading disable", () => {
    const component = renderer.create(<Loading />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
