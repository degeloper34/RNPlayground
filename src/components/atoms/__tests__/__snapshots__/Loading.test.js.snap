// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly with loading disable 1`] = `null`;

exports[`Loading renders correctly with loading enable 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.5)",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#F1D00A"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
